src/ZoneServer/World/Actors/Monsters/Mob.cs

```diff
@@ -381,47 +394,132 @@ private void DropItems(Character killer)
+					}
+
+					dropItem.Amount = rnd.Next(minAmount, maxAmount + 1);
+
+					if (killer == null || dropChance > autoloot)
+					{
+						var direction = new Direction(rnd.Next(0, 360));
+						var dropRadius = ZoneServer.Instance.Conf.World.DropRadius;
+						var distance = rnd.Next(dropRadius / 2, dropRadius + 1);
+
+						dropItem.SetLootProtection(killer, TimeSpan.FromSeconds(ZoneServer.Instance.Conf.World.LootPrectionSeconds));
+						dropItem.Drop(this.Map, this.Position, direction, distance);
+					}
+					else
+					{
+						killer.Inventory.Add(dropItem, InventoryAddType.PickUp);
+					}
 				}
+			}
+		}
 
-				var dropItem = new Item(itemData.Id);
-				var minAmount = dropItemData.MinAmount;
-				var maxAmount = dropItemData.MaxAmount;
+		/// <summary>
+		/// Rolls for a chance for this monster to become a rare (jackpot or elite) monster
+		/// </summary>
+		/// <param name="redOrb">true if this was spawned by a red orb and should get the boosted rates</param>
+		/// <returns></returns>
+		public void PossiblyBecomeRare(bool redOrb = false)
+		{
+			var rnd = RandomProvider.Get();
 
-				if (dropItemData.ItemId == ItemId.Silver || dropItemData.ItemId == ItemId.Gold)
-				{
-					minAmount = Math.Max(1, (int)(minAmount * (ZoneServer.Instance.Conf.World.SilverDropAmount / 100f)));
-					maxAmount = Math.Max(minAmount, (int)(maxAmount * (ZoneServer.Instance.Conf.World.SilverDropAmount / 100f)));
-				}
+			var worldConf = ZoneServer.Instance.Conf.World;
 
-				dropItem.Amount = rnd.Next(minAmount, maxAmount + 1);
+			var spawnRateBoost = 1F;
+			if (redOrb)
+				spawnRateBoost = worldConf.RedOrbJackpotRate / 100F;
 
-				if (killer == null || dropChance > autoloot)
-				{
-					var direction = new Direction(rnd.Next(0, 360));
-					var dropRadius = ZoneServer.Instance.Conf.World.DropRadius;
-					var distance = rnd.Next(dropRadius / 2, dropRadius + 1);
+			if (rnd.NextDouble() * 100D < worldConf.GoldJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.TwinkleBuff, 1, 0, TimeSpan.Zero, this);
+				return;
+			}
 
-					dropItem.SetLootProtection(killer, TimeSpan.FromSeconds(ZoneServer.Instance.Conf.World.LootPrectionSeconds));
-					dropItem.Drop(this.Map, this.Position, direction, distance);
-				}
-				else
-				{
-					killer.Inventory.Add(dropItem, InventoryAddType.PickUp);
-				}
+			if (rnd.NextDouble() * 100D < worldConf.BlueJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.SuperExp, 1, 0, TimeSpan.Zero, this);
+				return;
 			}
+
+			if (rnd.NextDouble() * 100D < worldConf.SilverJackpotSpawnChance * spawnRateBoost)
+			{
+				this.StartBuff(BuffId.SuperDrop, 1, 0, TimeSpan.Zero, this);
+				return;
+			}
+
+			// Check to see if this monster can become elite
+			// Note that on official it uses the map level rather than the monster level
+			if (this.Level < worldConf.EliteMinLevel)
+				return;
+
+			if (redOrb)
```

Comment:
It's commendable that you're always thinking about the customization potential and add the code for it, but I feel like this could be a little more flexible. Instead of essentially hardcoding a check for red orb monsters, which would only ever be useful for this one option, wouldn't be make more sense to let the caller pass in modified rates? This would decouple the code a little and open up the possibility of lowering or increasing the chances for other monster types.

Alternatively, another way to do this without hardcoding a relationship between these classes might be to simply let he monster figure out that it's a red orb monster, by setting a variable on it for example, and then apply the multipliers itself, removing the need for any special arguments.

Reply:
I actually did have it that way originally, but I changed it to this approach because it simplified the code slightly.  If you think any other code might ever want to override these rates I can return to having the rates be passed in.